src/app/content/utils/canonicalUrl.ts

```diff
@@ -22,7 +23,7 @@ export async function getCanonicalUrlParams(
 
     const canonicalBook = await getBook(id, version);
     const treeSection = findArchiveTreeNode(canonicalBook.tree, pageShortId);
-    if (treeSection) {
+    if (treeSection && hasOSWebData(canonicalBook)) {
```

Comment:
it would be fine for the unified book loader to only return `BookWithOSWebData`, and for that function to internally throw otherwise.